x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/datafeed_chart_flyout.tsx

```diff
@@ -154,21 +189,37 @@ export const DatafeedChartFlyout: FC<DatafeedChartFlyoutProps> = ({ jobId, end,
     setIsLoadingChartData(false);
   }, [endDate, data.bucketSpan]);
 
-  const getJobData = useCallback(async () => {
+  const getJobAndSnapshotData = useCallback(async () => {
     try {
       const job: CombinedJobWithStats = await loadFullJob(jobId);
+      const modelSnapshotResultsLine: LineAnnotationDatumWithModelSnapshot[] = [];
+      const modelSnapshotsResp = await getModelSnapshots(jobId);
+      const modelSnapshots = modelSnapshotsResp?.model_snapshots ?? [];
+      modelSnapshots.forEach((modelSnapshot) => {
+        const timestamp = Number(modelSnapshot?.latest_record_time_stamp);
+
+        modelSnapshotResultsLine.push({
+          dataValue: timestamp,
+          details: `${modelSnapshot.description}. ${
+            canCreateJob && canStartStopDatafeed ? revertSnapshotMessage : ''
```

Comment:
does something special happen if this `details` is an empty string? Is the annotation not displayed?
If so, it might be helpful to add a comment.

Reply:
It should not be an empty string ever since we're always adding the model snapshot description - which looks like it should always contain something. The annotation would still be displayed.

Reply:
Ah yes, I read that wrong.